_posts/2018-3-11-interactive-examples.md

```diff
@@ -0,0 +1,100 @@
+
+## Building a foundation
+
+After that not much happened for a while, because we didn't have any front-end resources. [Stephanie Hobson] helped improve the editor design, but she was also engaged in a full-scale redesign of MDN's article pages. In June [Schalk Neethling] joined the team, dedicated to this project. He built a solid foundation for the editors and a whole new contribution workflow. This would be the basis of the final implementation.
+
+## UX work and more user testing
+
+At the end of June, we showed the editors to [Jen Simmons] and [Dan Callahan], who provided us some very useful feedback. The JS editor seemed pretty good, but we were still having problems with the CSS editor. At this point it looked like this:
+
+![](../images/interactive-examples/css-editor-sf-all-hands.png)
+
+People didn't understand that they could edit the CSS, or even that the left-hand side consisted of a list of separate choices rather than a single block.
+
+Stephanie and Schalk did a full UX review of both editors. We also had an independent UX review from [Julia L-M] from the [Brigade]. After all this work, the editors looked like this:
+
+![](../images/interactive-examples/js-editor-final-user-test.png)
+
+![](../images/interactive-examples/css-editor-final-user-test.png)
+
+Then we had another round of user testing. This time we ran remote user tests over video, with participants recruited through MDN itself. This enabled us to have a tight feedback loop for the editors: we could quickly make and test adjustments based on user feedback.
+
+This time user testing was very positive, and we decided we were ready for beta testing.
+
+## Beta testing
+
+The beta test started at the end of August and lasted for two weeks. We embedded editors on three JS and three CSS pages, added a survey, and asked for feedback. [Danielle Vincent] mentioned it in the [MDN newsletter], which drove thousands of people to [our Discourse post announcing it].
+
+Feedback was overwhelmingly positive: 156/159 people who took the survey voted to see the editor on more pages, and the free-form text feedback was very encouraging. We were confident that we had a good UX.
+
+## JavaScript examples and page load optimization 
+
+Now we had an editor but very few actual examples. We asked [Mark Boas] to write examples for the JS reference pages, and in a couple of months he had written about 400 beautiful concise examples.
+
+<iframe width="100%" height="490px" frameborder="0" src="https://interactive-examples.mdn.mozilla.net/pages/js/array-slice.html"></iframe>
+
+We had another problem, though: it was clear that the editors regressed page load time too much. Schalk and Stephanie wrung every last millisecond of performance optimization out of the architecture, and finally, in December 2017, we decided to ship.
+
+In fact we're still not happy with the performance, but have some extra tricks planned for this year that we hope will make it back and more.
```

Comment:
Is there any small hint you can drop here about those 'tricks' or timing of the improvements? (to counterbalance the neg)

Reply:
This is basically about the CDN work. @atopal , can you help here?